src/Compilers/CSharp/Test/Emit3/RefStructInterfacesTests.cs

```diff
@@ -16718,6 +16718,108 @@ public class Helper
                 );
         }
 
+        [Fact]
+        public void IllegalBoxing_04()
+        {
+            var src = @"
+ref struct S
+{
+}
+
+public class Helper
+{
+    static object Test1<T>(T x)
+        where T : struct, allows ref struct
+    {
+        return x;
+    }
+
+    static object Test2(S x)
+    {
+        return x;
+    }
+
+    static object Test3<T>(T x)
+        where T : struct, allows ref struct
+    {
+        return (object)x;
+    }
+
+    static object Test4(S x)
+    {
+        return (object)x;
+    }
+}
+";
+            var comp = CreateCompilation(src, targetFramework: s_targetFrameworkSupportingByRefLikeGenerics);
+            comp.VerifyDiagnostics(
+                // (11,16): error CS0029: Cannot implicitly convert type 'T' to 'object'
+                //         return x;
+                Diagnostic(ErrorCode.ERR_NoImplicitConv, "x").WithArguments("T", "object").WithLocation(11, 16),
+                // (16,16): error CS0029: Cannot implicitly convert type 'S' to 'object'
+                //         return x;
+                Diagnostic(ErrorCode.ERR_NoImplicitConv, "x").WithArguments("S", "object").WithLocation(16, 16),
+                // (22,16): error CS0030: Cannot convert type 'T' to 'object'
+                //         return (object)x;
+                Diagnostic(ErrorCode.ERR_NoExplicitConv, "(object)x").WithArguments("T", "object").WithLocation(22, 16),
+                // (27,16): error CS0030: Cannot convert type 'S' to 'object'
+                //         return (object)x;
+                Diagnostic(ErrorCode.ERR_NoExplicitConv, "(object)x").WithArguments("S", "object").WithLocation(27, 16)
+                );
+        }
+
+        [Fact]
+        public void IllegalBoxing_05()
```

Comment:
>IllegalBoxing_05 [](http://example.com/codeflow?start=20&length=16)

Is this distinct from `IllegalBoxing_02`?

Reply:
> Is this distinct from `IllegalBoxing_02`?

It is true that some scenarios are duplicated, but I do not mind and do not plan to make any changes because of that.